### PR TITLE
Display study options properly on start on tablet

### LINKED ISSUE #8755: [Bug] Study option do not appear on start
###### Reproduction Steps

1. Open the app on a tablet
2. Check for study option (right side)

###### Expected Result

The selected deck's study option are shown

###### Actual Result

The study option remain empty

###### Debug info
Current master 425701e

### PR DESCRIPTION
**Pull request by ShridharGoel**
## Purpose / Description
Display study options properly on start on tablet.

## Fixes
Fixes #8755 

## Approach
`mFragmented` should be checked after it has been set in `onCreate()`.

## How Has This Been Tested?

Verified that the functionality is working.

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
Could you add a test for this?

Could you also confirm that the app still opens on a corrupt/upgradeable collection?

**Comment by ShridharGoel** [comment]
> Could you add a test for this?

Can it be done via Robolectric or an instrumentation test will be required (since this issue happens on tablets)?

> Could you also confirm that the app still opens on a corrupt/upgradeable collection?

How can this be checked?

**Comment by david-allison-1** [comment]
> > Could you add a test for this?
> 
> Can it be done via Robolectric or an instrumentation test will be required (since this issue happens on tablets)?
> 
> > Could you also confirm that the app still opens on a corrupt/upgradeable collection?
> 
> How can this be checked?

I've included a v16 apkg in the tests, change the DB path, close ankidroid and replace the folder's `collection.anki2`

Corruption? Copy the database file and blank one of the JSON fields in `col`

**Comment by david-allison-1** [comment]
Robolectric should be fine

**Comment by ShridharGoel** [comment]
I have added the test.

**Comment by ShridharGoel** [comment]
I have added a test.

File: AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.java [hunk 1]
```diff
@@ -407,6 +410,15 @@ public void notEnoughSpaceToBackupShowsError() {
         }
     }
 
+    @Test
+    @Config(qualifiers = "xlarge-port")
+    public void checkDisplayOfStudyOptionsOnTablet() {
+        DeckPickerEx deckPickerEx = super.startActivityNormallyOpenCollectionWithIntent(DeckPickerEx.class, new Intent());
+
+        StudyOptionsFragment studyOptionsFragment = (StudyOptionsFragment) deckPickerEx.getSupportFragmentManager().findFragmentById(R.id.studyoptions_fragment);
+        assertThat("Study options should show on start on tablet", studyOptionsFragment != null, is(true));
```
Comment (david-allison-1) [inline AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.java:419]:
```suggestion
        assertThat("Study options should show on start on tablet", studyOptionsFragment, notNullValue());
```

**Comment by david-allison-1** [comment]
@ShridharGoel Good to go after confirmation that you've tested this

**Comment by ShridharGoel** [comment]
This exception is coming in case of corrupt collection:

```
android.database.sqlite.SQLiteDatabaseCorruptException: error while compiling: "openAnkiDroidCollection": DBError { info: "SqliteFailure(Error { code: DatabaseCorrupt, extended_code: 11 }, Some(\"database disk image is malformed\"))", kind: Other }
        at net.ankiweb.rsdroid.BackendV1Impl.openAnkiDroidCollection(BackendV1Impl.java:135)
        at net.ankiweb.rsdroid.BackendMutex.openAnkiDroidCollection(BackendMutex.java:977)
        at net.ankiweb.rsdroid.BackendUtils.openAnkiDroidCollection(BackendUtils.java:27)
        at net.ankiweb.rsdroid.database.RustSupportSQLiteOpenHelper.createRustSupportSQLiteDatabase(RustSupportSQLiteOpenHelper.java:91)
        at net.ankiweb.rsdroid.database.RustSupportSQLiteOpenHelper.getWritableDatabase(RustSupportSQLiteOpenHelper.java:72)
        at com.ichi2.libanki.DB.<init>(DB.java:80)
        at com.ichi2.libanki.backend.RustDroidBackend.openCollectionDatabase(RustDroidBackend.java:42)
        at com.ichi2.libanki.Storage.Collection(Storage.java:78)
        at com.ichi2.anki.CollectionHelper.getCol(CollectionHelper.java:133)
        at com.ichi2.anki.CollectionHelper.getCol(CollectionHelper.java:115)
        at com.ichi2.anki.StudyOptionsFragment.getCol(StudyOptionsFragment.java:730)
        at com.ichi2.anki.StudyOptionsFragment.configureToolbarInternal(StudyOptionsFragment.java:381)
        at com.ichi2.anki.StudyOptionsFragment.configureToolbar(StudyOptionsFragment.java:371)
        at com.ichi2.anki.StudyOptionsFragment.onCreateView(StudyOptionsFragment.java:207)
        at androidx.fragment.app.Fragment.performCreateView(Fragment.java:2963)
        at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:518)
        at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:282)
        at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2189)
        at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2100)
        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:2002)
        at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:3136)
        at androidx.fragment.app.FragmentManager.dispatchActivityCreated(FragmentManager.java:3070)
        at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:251)
        at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:501)
        at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:246)
        at com.ichi2.anki.AnkiActivity.onStart(AnkiActivity.java:101)
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1425)
        at android.app.Activity.performStart(Activity.java:7825)
        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3294)
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
```

**Comment by ShridharGoel** [comment]
For corrupt collection case, it is working fine now.

> I've included a v16 apkg in the tests, change the DB path, close ankidroid and replace the folder's collection.anki2

I couldn't understand this properly. How to get the v16 apkg file?

**Comment by ShridharGoel** [comment]
For the corrupt collection case, it is working fine now.

> I've included a v16 apkg in the tests, change the DB path, close ankidroid and replace the folder's collection.anki2

I couldn't understand this properly. How to get the v16 apkg file?

**Comment by ShridharGoel** [comment]
@mikehardy This is almost complete. Can this also be included for release 2.15?

**Comment by david-allison-1** [comment]
> For the corrupt collection case, it is working fine now.
> 
> > I've included a v16 apkg in the tests, change the DB path, close ankidroid and replace the folder's collection.anki2
> 
> I couldn't understand this properly. How to get the v16 apkg file?

Sorry for the late reply: I'm trying a new method to manage my notifications

https://github.com/ankidroid/Anki-Android/blob/0cfc53fcbedbd3bf60bfa82ecc7aba94e3edcc3e/AnkiDroid/src/test/resources/schema16.anki2 should work

**Review by david-allison-1 — APPROVED** [review]
LGTM given testing worked

File: AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.java [hunk 2]
```diff
@@ -407,6 +410,15 @@ public void notEnoughSpaceToBackupShowsError() {
         }
     }
 
+    @Test
+    @Config(qualifiers = "xlarge-port")
```
Comment (mikehardy) [inline AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.java:414]: nice trick here on the `@Config` to add a test for it! Hadn't considered that

**Review by mikehardy — APPROVED** [review]
LGTM thank you - esp for the testing + confirms there

**Comment by mikehardy** [comment]
This is broken on tablets, it fails on release-2.15 branch immediately on startup when user has not granted permission.

I'm reverting the cherrypick to release-2.15 and 2.16alpha1 will not be released until fixed

Reproduction - use Pixel C emulator in landscape mode or portrait mode as first time user and open app. At least with commit hash 6a0b1d38e536c986c7531da4c45c467c49c59d66 it crashes immediately

**Pull request merged**